camayoc/tests/remote/rho/conftest.py

```diff
@@ -0,0 +1,149 @@
+# coding=utf-8
+"""Pytest customizations and fixtures for tests to execute on remote hosts."""
+
+import csv
+import os
+import pexpect
+import pytest
+import tempfile
+import shutil
+from io import BytesIO
+
+from camayoc import config
+from camayoc.tests.rho.utils import auth_add, input_vault_password
+from camayoc.utils import _XDG_ENV_VARS
+
+
+def pytest_generate_tests(metafunc):
+    """Do magic things with parameters.
+
+    This function enables us to create custom marks and do special things to
+    the parameters of fixtures based on the value of the marks.
+
+    """
+    if hasattr(metafunc.function, 'facts_needed'):
+        # this grabs the parameters handed to the pytest.mark.parametrize
+        # associated with the function that is marked 'facts_needed'
+        profiles_to_scan = metafunc.function.parametrize
+        # now we will pass those parameters to another function
+        # to perform the scan and collect all testable facts.
+        # This will have the effect of the test being parametrized
+        # on the facts, not the profiles, so a test will run
+        # for each fact
+        metafunc.function.parametrize = [testable_facts(profiles_to_scan)]
+
+
+def testable_facts(profiles_to_scan):
+    """Perform scans on each profile and return facts.
+
+    This function will attempt to scan all profiles listed in config file. For
+    those that are successful, facts will be returned so the test may be
+    parametrized on them.
+
+    Additional data about the facts like what host, profile, auth, actual and
+    expected value are collected into a single dictionary with the scanned fact
+    and passed to the test that is marked as follows:
+
+        @pytest.mark.facts_needed
+        @pytest.mark.parametrize('fact', profiles())
+
+    """
+    profiles_to_scan = profiles_to_scan.args[1]
+    cfg = config.get_config()
+    facts_to_test = []
+    for profile in profiles_to_scan:
+        cwd = os.getcwd()
```

Comment:
It seems that you are duplicating what the isolated filesystem does here, so I think it would be better to use the context manager here https://github.com/quipucords/camayoc/blob/master/camayoc/utils.py#L20

Reply:
Thanks for bringing this up, when I realized I could not use the fixture did not connnect the dots that I could just use the context manager.